src/biz/ChartRenderer.js

```diff
@@ -49,6 +39,20 @@ class ChartRenderer {
   }
 
 
+  /** Generates line colors. One gradient per drivetrain. One color per gear. */
+  generateLineColors(drivetrains) {
```

Comment:
What's the story with { gearRatios }? Does that somehow stuff the collection in place of what would normally be the "current" object? Is there a term for this?

Reply:
Object destructuring... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring

Reply:
When I use gradients.push() or the map function laid out above, the result is an array of arrays, instead of an array:

[[000000, ffffff], [000000, ffffff]] vs [000000, ffffff, 000000, ffffff]

It's...kind of a weird data format that the chart demands, which is why all this is in its own wrangling class.

I still can't find a better way to do this while maintaining a "flat" array of values...

Reply:
Aha that makes sense.